### PR TITLE
Add test to assert CHANGELOG format

### PR DESCRIPTION
**Pull request by jderusse**
This PR adds 2 tests:
- Assert that a CHANGELOG entry is present
- Assert the CHANGELOG files content are consistent

all files MUST follow the following format (inspired by https://keepachangelog.com/en/1.1.0/)
```
# Change Log

## NOT RELEASED

## 1.2.0

### Added|Changed|Fixed|Deprecated|Removed|Security

- single line entry
- multiline
  entry

## 1.1.0

- Empty release

## 0.1

First version

```

It also fixes all previous change log files to make the baseline.

### CODE REVIEW

File: src/Integration/Laravel/Queue/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## 0.2.1
 
-### Changed
+### Fixed
```
Comment (stof) [inline src/Integration/Laravel/Queue/CHANGELOG.md:7]: Why not allowing `Changed` anymore ? Not all changes are bug fixes.

File: tests/Unit/ChangelogTest.php [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+class ChangelogTest extends TestCase
+{
+    /**
+     * @dataProvider provideChangedlogFiles
+     */
+    public function testChangelogFormat(string $changelogPath)
+    {
+        $lines = explode("\n", file_get_contents($changelogPath));
+
+        self::assertGreaterThan(3, \count($lines), 'CHANGEOG must not be empty');
```
Comment (stof) [inline tests/Unit/ChangelogTest.php:22]:
```suggestion
        self::assertGreaterThan(3, \count($lines), 'CHANGELOG must not be empty');
```

File: tests/Unit/ChangelogTest.php [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                $this->expectNotToPerformAssertions();
+
+                return;
+            }
+        }
+
+        self::fail('Missing CHANGELOG entry for package ' . $service);
+    }
+
+    public function provideChangedlogFiles(): iterable
```
Comment (stof) [inline tests/Unit/ChangelogTest.php:162]: Data providers should be static methods. Non-static ones are deprecated in latest PHPUnit.

File: tests/Unit/ChangelogTest.php [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+            if (0 === strpos($line, '## ')) {
+                self::assertTrue($wasEmptyLine, 'CHANGELOG line ' . ($index - 1) . ' MUST be empty');
+                self::assertTrue($inTitle1, 'CHANGELOG line ' . $index . ' MUST have title2 inside a title1');
+                self::assertFalse($title2Empty, 'CHANGELOG line ' . $index . ' MUST not have empty title2');
+                self::assertFalse($title3Empty, 'CHANGELOG line ' . $index . ' MUST not have empty title3');
+                self::assertMatchesRegularExpression('/^## (NOT RELEASED|\d+\.\d+\.\d+)$/', $line, 'CHANGELOG line ' . $index . ' MUST be "## X.Y.Z"');
+                $inTitle2 = true;
+                $inTitle3 = false;
+                $title3s = [];
```
Comment (stof) [inline tests/Unit/ChangelogTest.php:71]: Before resetting the list of title 3, should we assert the order of title 3 of the previous section to ensure consistent order.

Comment (jderusse) [reply tests/Unit/ChangelogTest.php:71]: Yeah, I hesitated about this. But fear the diff to fix the existing changelogs...


let's do it once for all

File: tests/Unit/ChangelogTest.php (hunk 3 above)
Comment (jderusse) [reply tests/Unit/ChangelogTest.php:162]: thanks

File: src/Integration/Laravel/Queue/CHANGELOG.md (hunk 1 above)
Comment (jderusse) [reply src/Integration/Laravel/Queue/CHANGELOG.md:7]: I would like to have as few "words" as possible

- `BC-Break` and `Added` works well for Major and Minor.
- 'Fixed` works well to.
- for Patches, do you think `Changes` is not ambiguous? (like "I changed the method signatures")

Should we allow `Chores`, `Perf`, `Refactor`, ... like in https://www.conventionalcommits.org/en/v1.0.0/ ?

Comment (jderusse) [reply src/Integration/Laravel/Queue/CHANGELOG.md:7]: Let's go with https://keepachangelog.com/en/1.1.0/

**Pull request merged**